libethcore/TransactionReceipt.cpp

```diff
@@ -29,35 +29,34 @@ using namespace dev::eth;
 TransactionReceipt::TransactionReceipt(bytesConstRef _rlp)
 {
     RLP r(_rlp);
+    m_status = static_cast<eth::TransactionException>((uint32_t)r[4]);
     m_stateRoot = (h256)r[0];
     m_gasUsed = (u256)r[1];
     m_contractAddress = (Address)r[2];
     m_bloom = (LogBloom)r[3];
-    m_status = static_cast<eth::TransactionException>((uint32_t)r[4]);
     m_outputBytes = (bytes)r[5];
-    for (auto const& i : r[6])
-        m_log.emplace_back(i);
+    decodeLog(r);
 }
 
 TransactionReceipt::TransactionReceipt(h256 const& _root, u256 const& _gasUsed,
     LogEntries const& _log, eth::TransactionException _status, bytes _bytes,
     Address const& _contractAddress)
 {
+    m_status = _status;
```

Comment:
Codacy found an issue: [Variable 'm_status' is assigned in constructor body. Consider performing initialization in initialization list.](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=6313922)